bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/LspService.java

```diff
@@ -47,25 +47,37 @@ public void restartLspServers()
 
 	public void updateAdditionalOptions(String additionalOptions)
 	{
+		if (additionalOptions == null)
+		{
+			return;
+		}
 		String qualifier = configuration.qualifier();
 		InstanceScope.INSTANCE.getNode(qualifier).put(ClangdMetadata.Predefined.additionalOptions.identifer(),
 				additionalOptions);
 	}
 
 	public void updateLspQueryDrivers()
 	{
+		String toolchainPath = IDFUtil.getToolchainExePathForActiveTarget();
 		String qualifier = configuration.qualifier();
+		if (toolchainPath == null)
+		{
+			return;
+		}
 		// By passing --query-driver argument to clangd helps to resolve the
 		// cross-compiler toolchain headers.
-		String toolchainPath = IDFUtil.getToolchainExePathForActiveTarget();
 		InstanceScope.INSTANCE.getNode(qualifier).put(ClangdMetadata.Predefined.queryDriver.identifer(), toolchainPath);
 	}
 
 	public void updateClangdPath()
 	{
+		String clangdPath = IDFUtil.findCommandFromBuildEnvPath(ILSPConstants.CLANGD_EXECUTABLE);
+		if (clangdPath == null)
+		{
+			return;
```

Comment:
Log the message so we're aware that clangdPath was null and the path was configured - useful for troubleshooting purposes and same goes for other places as well.